include/zephyr/toolchain/gcc.h

```diff
@@ -14,6 +14,15 @@
  * Macros to abstract compiler capabilities for GCC toolchain.
  */
 
+/*
+ * gcc.h is included indirectly from devicetree.h and uses autoconf.h.
+ * Non-zephyr builds such as external bootloaders and secure
+ * root-of-trust images need to include devicetree.h but may not be
+ * using gcc's -include autoconf.h so we add a redundant explicit
+ * include here as well.
+ */
+#include <autoconf.h>
```

Comment:
why is `<autoconf.h>` needed? See https://github.com/zephyrproject-rtos/zephyr/pull/44471

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/50852#discussion_r984570024